src/main/flight/gps_rescue.c

```diff
@@ -263,15 +274,19 @@ static void rescueAttainPosition()
     // Speed and altitude controller internal variables
     static float previousSpeedError = 0;
     static int16_t speedIntegral = 0;
-    static float previousAltitudeError = 0;
-    static int16_t altitudeIntegral = 0;
+    static int16_t zVelocityError = 0;
```

Comment:
I don't see any reason for this to be a `static`. It gets reassigned every loop and there's no use of it's previous value.

Also there's really no reason to constrain these values to `int16` and why not just use `int`? Performance is better because it avoids the implicit conversion before math anyway. Also avoid possible overflow/underflow issues for unexpected values. Also look at the derivative calc.